### PR TITLE
Fix Jakarta Cookie serialization in native

### LINKED ISSUE #35582: Unable to serialize jakarta.ws.rs.core.Cookie after updating to 3.2.0 in native-image
### Describe the bug

After updating to Quarkus I am unable to serialize objects containing `jakarta.ws.rs.core.Cookie` when running in native-image.
I am using the `quarkus-resteasy-reactive-jackson` dependency. I added a custom serializer to mitigate and the error has gone.

I get the error with macOS and Linux native-images.

### Expected behavior

The Class `jakarta.ws.rs.core.Cookie` is serialized to json.

### Actual behavior

I am getting an error during serialization:

```
2023-08-27 04:47:14,717 ERROR [io.qua.res.rea.jac.run.map.NativeInvalidDefinitionExceptionMapper] (executor-thread-1) Jackson was unable to serialize type 'jakarta.ws.rs.core.Cookie'. Consider annotating the class with '@RegisterForReflection' or using 'org.jboss.resteasy.reactive.RestResponse' as a response type of 'de.afrouper.server.EchoResource#get: com.fasterxml.jackson.databind.exc.InvalidDefinitionException: No serializer found for class jakarta.ws.rs.core.Cookie and no properties discovered to create BeanSerializer (to avoid exception, disable SerializationFeature.FAIL_ON_EMPTY_BEANS). This appears to be a native image, in which case you may need to configure reflection for the class that is to be serialized (through reference chain: de.afrouper.server.dto.EchoReply["request"]->de.afrouper.server.dto.Request["cookies"]->java.util.Collections$UnmodifiableMap["X-OPENHAB-SESSIONID"])
```

### How to Reproduce?

_No response_

### Output of `uname -a` or `ver`

_No response_

### Output of `java -version`

OpenJDK Runtime Environment GraalVM CE 22.3.0 (build 17.0.5+8-jvmci-22.3-b08)

### GraalVM version (if different from Java)

OpenJDK Runtime Environment GraalVM CE 22.3.0 (build 17.0.5+8-jvmci-22.3-b08)

### Quarkus version or git rev

3.2.0

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

Apache Maven 3.9.4

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by geoand**
Fixes: #35582

### CODE REVIEW

File: extensions/resteasy-reactive/quarkus-resteasy-reactive-jackson/deployment/src/main/java/io/quarkus/resteasy/reactive/jackson/deployment/processor/ResteasyReactiveJacksonProcessor.java [hunk 1]
```diff
@@ -205,6 +206,11 @@ private String getJacksonMessageBodyReader(boolean needsFullFeatureSet) {
                 : ServerJacksonMessageBodyReader.class.getName();
     }
 
+    @BuildStep
+    void reflection(BuildProducer<ReflectiveClassBuildItem> producer) {
+        producer.produce(ReflectiveClassBuildItem.builder(Cookie.class).methods().build());
```
Comment (gsmet) [inline extensions/resteasy-reactive/quarkus-resteasy-reactive-jackson/deployment/src/main/java/io/quarkus/resteasy/reactive/jackson/deployment/processor/ResteasyReactiveJacksonProcessor.java:211]: Is it really expected that you have to serialize a `Cookie`?

Comment (geoand) [reply extensions/resteasy-reactive/quarkus-resteasy-reactive-jackson/deployment/src/main/java/io/quarkus/resteasy/reactive/jackson/deployment/processor/ResteasyReactiveJacksonProcessor.java:211]: There is no reason not to really. We don't to it ourselves, but I don't see any reason why we should make it more difficuklt

Comment (gsmet) [reply extensions/resteasy-reactive/quarkus-resteasy-reactive-jackson/deployment/src/main/java/io/quarkus/resteasy/reactive/jackson/deployment/processor/ResteasyReactiveJacksonProcessor.java:211]: OK, I consider it an internal data structure but yeah why not.

Comment (geoand) [reply extensions/resteasy-reactive/quarkus-resteasy-reactive-jackson/deployment/src/main/java/io/quarkus/resteasy/reactive/jackson/deployment/processor/ResteasyReactiveJacksonProcessor.java:211]: Yeah, it's kinda grey :)

**Review by gastaldi — APPROVED** [review]

**Pull request merged**